zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/BackupApiRequestHandlerStep.java

```diff
@@ -48,16 +48,8 @@ public String getName() {
 
   private ActorFuture<Void> installRequestHandler(final PartitionTransitionContext context) {
     final ActorFuture<Void> installed = context.getConcurrencyControl().createFuture();
-    final var writerFuture = context.getLogStream().newLogStreamWriter();
-    writerFuture.onComplete(
-        (logStreamRecordWriter, error) -> {
-          if (error == null) {
-            createBackupApiRequestHandler(context, installed, logStreamRecordWriter);
-          } else {
-            installed.completeExceptionally(error);
-          }
-        });
-
+    final var logStreamRecordWriter = context.getLogStream().newLogStreamWriter();
```

Comment:
:question: Same as above, what happens here if the logstream is closed?

Reply:
This is executed by the `ZeebePartition` actor, which handles all failed actor jobs in `onInstallFailure`. In our case this ends up at `handleRecoverableFailure` which ultimately steps down and makes the partition inactive.

This is the same that would have happened before. The install future would have completed exceptionally, which `ZeebePartition#leaderTransition` handles by calling `onInstallFailure`.

This is now less explicit than before so let me know if you'd prefer a try-catch block to explicitly complete the future exceptionally instead of going through `ZeebePartition#handleFailure`.

Reply:
To be fair, it's something I wasn't sure about either and I should have checked it before opening the PR :sweat_smile: